src/header/TransferBench.hpp

```diff
@@ -3188,6 +3189,39 @@ static bool IsConfiguredGid(union ibv_gid const& gid)
     }
   }
 
+  // Dispatch wrapper: Selects specialized kernel based on runtime numSrcs/numDsts
+  template <typename PACKED_FLOAT, int BLOCKSIZE, int UNROLL, int TEMPORAL_MODE>
+  __global__ void __launch_bounds__(BLOCKSIZE)
+    GpuReduceKernel(SubExecParam* params, int seType, int warpSize, int waveOrder, int numSubIterations)
+  {
+    // Read numSrcs and numDsts from params
+    int const numSrcs = params[blockIdx.y].numSrcs;
+    int const numDsts = params[blockIdx.y].numDsts;
+    
+    // Dispatch to specialized implementation for common cases
+    if (numSrcs == 1 && numDsts == 1) {
```

Comment:
Use if (1,1) / else if (1,0) / else if (0,1) / else (-1,-1) and skip the returns.